src/Symfony/Component/DependencyInjection/Dumper/XmlDumper.php

```diff
@@ -116,9 +116,14 @@ private function addService(Definition $definition, ?string $id, \DOMElement $pa
         if ($definition->isLazy()) {
             $service->setAttribute('lazy', 'true');
         }
-        if (null !== $decorated = $definition->getDecoratedService()) {
-            list($decorated, $renamedId, $priority) = $decorated;
+        if (null !== $decoratedService = $definition->getDecoratedService()) {
+            list($decorated, $renamedId, $priority) = $decoratedService;
             $service->setAttribute('decorates', $decorated);
+            $onDecoratedInvalid = $decoratedService[3] ?? ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE;
+            if (ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE !== $onDecoratedInvalid) {
+                $invalidBehavior = ContainerInterface::NULL_ON_INVALID_REFERENCE === $onDecoratedInvalid ? 'null' : 'ignore';
```

Comment:
defensive programming here too: null / ignore should be dumped only when the match is confirmed, and any other value should not generate any dump